### PR TITLE
[Test] Fix flaky object spilling test

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This works. Need to revert the skipped tests.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/core_worker/core_worker.cc [hunk 1]
```diff
@@ -686,6 +689,14 @@ void CoreWorker::Shutdown() {
   }
 }
 
+void CoreWorker::NotifyRaylet() {
```
Comment (ericl) [inline src/ray/core_worker/core_worker.cc:692]: nit: can we rename this to ConnectToRaylet?

File: src/ray/core_worker/core_worker.cc [hunk 2]
```diff
@@ -686,6 +689,14 @@ void CoreWorker::Shutdown() {
...
+  RAY_CHECK(!options_.notify_automatically);
```
Comment (ericl) [inline src/ray/core_worker/core_worker.cc:693]:
```suggestion
  RAY_CHECK(!options_.connect_on_start);
```

File: src/ray/core_worker/core_worker.cc (hunk 1 above)
Comment (rkooo567) [reply src/ray/core_worker/core_worker.cc:692]: Technically, this function doesn't "connect" to raylet, and that's why I chose this name (connection happens when the client class is created, which happens in the constructor). But I think it is still okay to use this name. I will follow your suggestion, but if you have a better name in mind, plz let me know!

**Review by ericl — COMMENTED** [review]
Ah nevermind

**Review by ericl — APPROVED** [review]

**Pull request merged**